frontend/lib/verification/view/verification_view.dart

```diff
@@ -63,16 +64,43 @@ class _VerificationViewState extends State<VerificationView> {
   }
 
   void _onScanCodePressed() {
-    final provider =
-        Provider.of<VerificationCardDetailsModel>(context, listen: false);
-
     Navigator.push(
         context,
         MaterialPageRoute(
-          builder: (context) => QRCodeScanner(
-            qrCodeContentParser:
-                VerificationProcessor(provider).processQrCodeContent,
+          builder: (context) => QrCodeScannerPage(
+            title: "Karte verifizieren",
+            onCodeScanned: _handleQrCode,
           ),
         ));
   }
+
+  Future<void> _handleQrCode(String rawQRContent) async {
+    // TODO still no beautiful function
+    final provider =
+    Provider.of<VerificationCardDetailsModel>(context, listen: false);
+    void handleError(String message, String errorCode, Exception e) {
+      print("Verification failed: ${e.toString()}");
+      provider.setVerificationFailure(VerificationError(message,
+          errorCode));
+    }
```

Comment:
wait, a function in a function? 🤔

Reply:
I found this a bit shorter than putting print/setFailure commands in every catch clause